commit-message.md

```diff
@@ -0,0 +1,36 @@
+# Guidelines to create great commit messages
+
+
+We use [GitCop](https://gitcop.com) to check that commit messages are
+properly written. The rules are the following:
+
+* The first line of a commit message, called the subject line should
+  not be more than 80 characters long.
+
+* The commit message should end with the following trailers:
+
+  ```
+  Licence: MIT
+  Signed-off-by: User Name <email@address>
+  ```
+
+  where "User Name" is the author's real name and email@address one of
```

Comment:
The DCO and copyright licenses are for ensuring protections + rights to use / modify code are respected. These only matter in various nation's legal courts, and these names are only relevant in that context. I am not sure how other projects handle identity, but pseudonyms/personas should be fine according to most legal codes (it's fine in US code and lots of other copyright systems are based on or compatible with the US one).

However, the linux kernel contribution guidelines say:
> then you just add a line saying
>     Signed-off-by: Random J Developer <random@developer.example.org>
> using your real name (sorry, no pseudonyms or anonymous contributions.)
> -- from https://www.kernel.org/doc/Documentation/SubmittingPatches

I presume this is for both security + legal concerns (It may be tricky to deal with pseudonymous contribution in other countries. And it certainly is easier to trust contributions if you know who wrote them, and why.) I'm sure some pseudonymous contribution case has come up though-- i'd be curious how they handled it.

For our purposes, until further notice, i'm fine accepting pseudonymous contributions that bear the `License: MIT` trailer, so that it is known -- beyond shadow of a doubt -- that the license has been properly signed off to.

---

one the internet nobody know's i'm a cat :)

![](http://gateway.ipfs.io/ipfs/QmTqXR188uyYSZwHDXb44cQv6PYQhgCyzMxEQh65wmV7hv/icat.jpg)